docs/guides/grpc.md

```diff
@@ -0,0 +1,259 @@
+    repeated NewsId ids = 1;
+}
+
+// List of all news
+message NewsList {
+   repeated News news = 1;
+}
+
+// NewsService defines read and write operations for news items
+service NewsService {
+    // GetAllNews retrieves all news items without any arguments
+    rpc GetAllNews (google.protobuf.Empty) returns (NewsList) {}
+
+    // GetNews fetches a single news item by its ID
+    rpc GetNews (NewsId) returns (News) {}
+
+    // GetMultipleNews retrieves multiple news items based on their IDs
+    rpc GetMultipleNews (MultipleNewsId) returns (NewsList) {}
+}
+```
+
+### Implement gRPC service
+
+Now having the protobuf file you can write a server that implements `NewsService` at any language you want that supports gRPC.
+
+Tailcall organization has sample node.js service inside [this repo](https://github.com/tailcallhq/node-grpc) that you can pull to your local machine
+
+To spin up the sample service just run inside repo
+
+```sh
+npm i
+node server
+```
+
+and wait for logs about service is running.
+
+## Tailcall config
+
+Now when we have a running gRPC service we're going to write Tailcall's config to make the integration.
+
+To do this we need to specify GraphQL types corresponding to gRPC types we have defined in protobuf file. Let's create new file `grpc.graphql` file with following content:
+
+```graphql
+# The GraphQL representation for News message type
+type News {
+  id: Int
+  title: String
+  body: String
+  postImage: String
+}
+
+# Input type that is used to fetch news data by its id
+input NewsInput {
+  id: Int
+}
+
+# Resolves multiple news entries
+type NewsData {
+  news: [News]!
+}
+```
+
+Now when we have corresponding types in schema we want to define GraphQL Query that specifies the operation we can execute onto news. We can extend our config with next Query:
+
+```graphql
+type Query {
+  # Get all news i.e. NewsService.GetAllNews
+  news: NewsData!
+  # Get single news by id i.e. NewsService.GetNews
+  newsById(news: NewsInput!): News!
+}
+```
+
+It not working example yet because we haven't specified how to use connect to server. To do this you may want to explore more about [`@grpc` operator](../operators/grpc.md). It's usage is pretty straightforward and require you to specify some required options like path to protobuf file itself, name of the service and method that should be used to make a call.
+
+If you need to provide any input to gRPC method call you can specify it with the `body` option that allows to specify a Mustache template and therefore it could use any input data like `args` and `value` to construct the body request. The body value is specified in the JSON format if you need to create the input manually and cannot just use `args` input.
+
+```graphql
+type Query {
+  news: NewsData! @grpc(service: "NewsService", method: "GetAllNews", protoPath: "news.proto")
+  newsById(news: NewsInput!): News!
+    @grpc(service: "NewsService", method: "GetNews", body: "{{args.news}}", protoPath: "news.proto")
```

Comment:
Need to change this and add the `@link(type: Protobuf)` to import the .proto file
```suggestion
  news: NewsData! @grpc(method: "news.NewsService.GetAllNews")
  newsById(news: NewsInput!): News!
    @grpc(method: "news.NewsService.GetNews", body: "{{args.news}}")
```

Reply:
updated